### ISSUE TITLE
parsing of Born effective charges chokes database insertion

### ISSUE THREAD

**Issue by computron** [materialsproject/pymatgen#373]
## System

* Pymatgen version: master
* Python version: 2.7

## Summary

* The parsing of the born effective charges takes place around lines: pymatgen/io/vasp/outputs.py:1833
* The format is a dictionary, where keys are *int* of atom index, and values are the BEC matrix
* For example it looks like this:
```
u'born': {0: array([[ -9.58400000e-02,   5.50000000e-04,   3.90000000e-04],
       [  0.00000000e+00,  -9.52700000e-02,   6.00000000e-05],
       [ -0.00000000e+00,  -0.00000000e+00,  -9.60900000e-02]]), 1: array([[ -9.58400000e-02,  -5.50000000e-04,  -3.90000000e-04],
       [  0.00000000e+00,  -9.64400000e-02,  -7.00000000e-05],
       [  0.00000000e+00,  -0.00000000e+00,  -9.56000000e-02]])}
```
* This chokes any Drone that tries to enter this data into the database, since MongoDB does not allow int keys (only str)

## Suggested solution (if any)

* Either use string keys, or change the format entirely - e.g. array of tensor rather than dict of tensor, or array of dict where each member sub-dict has an ion_idx and born keys.

More information available if needed...

**Comment by computron** [comment]
jsanitizing the output is another option...

**Comment by computron** [comment]
I am pretty sure I tried this exact solution but got some different serialization error since there was numpy arrays embedded in the dict. The jsanitize() would surely clear it up, but the raw dict still had errors.

**Comment by computron** [comment]
(I also made an attempt to turn the numpy matrices into lists - but again hit some error which is when I stopped working on it).

**Comment by shyuep** [comment]
Not quite sure what's the problem with turning numpy matrices into lists. Isn't it just array.tolist()? That always works. 

I didn't bother to convert the numpy array in the class itself because there are a lot of instances where you actually prefer the numpy array (e.g., to do mass manipulation effficiently). For insertion into DBs, try the tolist() method.

**Comment by computron** [comment]
I didn't use to_list(), but I did try avoiding the numpy array entirely for the Born charges by having it natively be parsed as a list. I ended up getting an int index error for db insertion like before, which didn't make much sense to me but I also didn't try looking into it more.

Anyway, since you have made the new jsanitize() and fixed the format of this, I will just work with that for now and speak up if I run into any more issues.
